lib/providers/RidesProvider.dart

```diff
@@ -13,8 +13,8 @@ import 'AuthProvider.dart';
 /// Manage the state of rides with ChangeNotifier.
 class RidesProvider with ChangeNotifier {
   Ride currentRide;
-  List<Ride> pastRides = [];
-  List<Ride> upcomingRides = [];
+  List<Ride> pastRides;
+  List<Ride> upcomingRides;
```

Comment:
How does changing this from an initialized empty list to null impact refreshing?

Reply:
It's used to help with checking if we've fetched to prevent flashing the empty ride widgets before it finishes fetching; we have to use null instead of empty list because we don't know if there actually just aren't rides